### PR TITLE
[wheel] Add support for manylinux python3.8

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
A portion of #15959.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16226)

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
@drake-jenkins-bot linux-focal-unprovisioned-gcc-wheel-experimental-snopt-release please

**Comment by jwnimmer-tri** [comment]
This is passing for me locally.  We'll see what Jenkins has to say.

+@mwoehlke-kitware for feature review, please.

File: tools/wheel/build-wheels [hunk 1]
```diff
@@ -217,7 +216,11 @@ def test_wheel(target, options):
     Runs the test script for the wheel matching the specified target.
     """
     vm = f'cp{target.python_version}'
-    wheel = f'drake-{options.version}-{vm}-{vm}m-manylinux_2_27_x86_64.whl'
+    if target.python_version < '38':
+        abi = 'm'
+    else:
+        abi = ''
+    wheel = f'drake-{options.version}-{vm}-{vm}{abi}-manylinux_2_27_x86_64.whl'
```
Comment (mwoehlke-kitware) [inline tools/wheel/build-wheels:223]: (Sure would be nice if Reviewable had Suggestions... Or the ability to comment on multiple lines...)

Strictly speaking, I think this:
```suggestion
    if target.python_version < '38':
        abi = f'{vm}m'
    else:
        abi = f'{vm}'
    wheel = f'drake-{options.version}-{vm}-{abi}-manylinux_2_27_x86_64.whl'
```
...might be more accurate? (As a bonus, it makes the `wheel =` line shorter 🙂.)

**Review by jwnimmer-tri — COMMENTED** [review]
+@ggould-tri for platform review per schedule, please.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16226#-:-MqzNboSBwWTRxcJClgI:b4mx0my)* status: 1 unresolved discussion, LGTM missing from assignees mwoehlke-kitware,ggould-tri(platform) (waiting on @ggould-tri and @mwoehlke-kitware)

___
*[tools/wheel/build-wheels, line 223 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16226#-Mqz135vViXWfpCw_Hxu-r1-223:-MqzNXi1Db0hmuwnOCcY:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/b95d0d774ff12a97f8f1c3117f02f00e12fce23f/tools/wheel/build-wheels#L223)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

(Sure would be nice if Reviewable had Suggestions... Or the ability to properly comment on multiple lines...)

Strictly speaking, I think this:
```suggestion
    if target.python_version < '38':
        abi = f'{vm}m'
    else:
        abi = f'{vm}'
    wheel = f'drake-{options.version}-{vm}-{abi}-manylinux_2_27_x86_64.whl'
```
...might be more accurate? (As a bonus, it makes the `wheel =` line shorter 🙂.)
</blockquote></details>

Done.

**Comment by jwnimmer-tri** [comment]
@drake-jenkins-bot linux-focal-unprovisioned-gcc-wheel-experimental-snopt-release please

**Review by mwoehlke-kitware — COMMENTED** [review]
:lgtm: (provided CI is happy)

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16226#-:-MqzPsyOCOSJgjpVhrEa:b-oy65nb)* status: LGTM missing from assignee ggould-tri(platform) (waiting on @ggould-tri)

**Review by ggould-tri — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16226#-:-MqzVea58vkL5akVv17p:bnfp4nl)* status: :shipit: complete! all discussions resolved, LGTM from assignees ggould-tri(platform),mwoehlke-kitware (waiting on @jwnimmer-tri)

**Comment by jwnimmer-tri** [comment]
CI is happy with all three:
```
[2021-12-15T19:09:02.575Z] /opt/python/lib/python3.6/site-packages/pydrake/share/drake
[2021-12-15T19:14:27.604Z] /opt/python/lib/python3.7/site-packages/pydrake/share/drake
[2021-12-15T19:19:27.726Z] /opt/python/lib/python3.8/site-packages/pydrake/share/drake
```

**Pull request merged**